app/views/admin/users/index.html.haml

```diff
@@ -17,7 +17,9 @@
       %tr
         %td= user.email
         %td.actions
-          = link_to "Edit", edit_admin_user_path(user), class: "btn btn-small"
-          = link_to "Delete", admin_user_path(user), method: :delete, data: { confirm: "Are you sure you want to delete #{user}?" }, class: "btn btn-small"
+          - if policy(user).edit?
```

Comment:
Prefer not - what benefits do you see? I think part of not being able to do something is not knowing that the something is even an option.